databricks_cli/tunnel/api.py

```diff
@@ -0,0 +1,226 @@
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from pathlib import Path
+from typing import Tuple
+
+import requests
+import tornado
+from cryptography.hazmat.primitives.asymmetric import rsa
+from tornado.ioloop import IOLoop
+
+from databricks_cli.sdk import ClusterService
+from databricks_cli.sdk.v1_service import CommandExecutionService
+from databricks_cli.tunnel.server import TunnelConfig, StreamingServer
+
+
+def generate_key_pair() -> Tuple[bytes, bytes]:
+    from cryptography.hazmat.primitives import serialization
+
+    key = rsa.generate_private_key(
+        public_exponent=65537,
+        key_size=2048
+    )
+
+    private_key = key.private_bytes(
+        serialization.Encoding.PEM,  # noqa
+        serialization.PrivateFormat.TraditionalOpenSSL,  # noqa
+        serialization.NoEncryption()
+    )
+
+    public_key = key.public_key().public_bytes(
+        serialization.Encoding.OpenSSH,  # noqa
+        serialization.PublicFormat.OpenSSH  # noqa
+    )
```

Comment:
Are these the same parameters that `ssh-keygen` would use when generating a key pair? For e.g., GitHub's recommendation is `ssh-keygen -t rsa -b 4096 ...` so we want to make sure we're using the recommended defaults here

Reply:
no... another thing that i notice is we don't have the user and the hostname. these aren't required, but usually there when we generate ssh keys. cc: @marcom-db